### ISSUE TITLE
Error when passing model/data to break_down function

### ISSUE THREAD

**Issue by hbaniecki** [ModelOriented/iBreakDown#43]
From break_down() examples:
This throws a note:
```
library("iBreakDown")
library("DALEX")
library("randomForest")
set.seed(1313)

model <- randomForest(status ~ . , data = HR)
new <- HR_test[1,]

explainer_rf <- explain(model,
                        data = HR[1:1000,1:5],
                        y = HR$status[1:1000])
```
>Please note that 'y' is a factor.[...]

This works:
```
break_down(explainer_rf, new)
```

This throws an error:
```
break_down(x=model, data = HR[1:1000,1:5], predict_function = predict, new_observation = new)
```
> Error in break_down.default(x = model, data = HR[1:1000, 1:5], new_observation = new,  : 
  promise already under evaluation: recursive default argument reference or earlier problems?

This throws an error:
```
local_attributions(x=model, data = HR[1:1000,1:5], predict_function = predict, new_observation = new)
```
> Error in colMeans(yhatpred) : 'x' must be numeric

**Comment by pbiecek** [comment]
I guess that the problem is in the `predict` function for randomForest.
For classification by default the predict.randomForest returns classes not scores and other functions cannot calculate averages from classes.

DALEX has overloaded `yhat` function for prediction, and the `DALEX:::yhat.randomForest` recognizes classification forests. In such case it is using `pred <- predict(X.model, newdata, type = "prob", ...)`.

So this example shows that it is better to use DALEX adapters, because otherwise user needs to define own `predict_fucntion` for some models

**Comment by hbaniecki** [comment]
This example shows the real problem:
```
library("iBreakDown")
library("DALEX")

titanic <- na.omit(titanic)
set.seed(1313)
titanic_small <- titanic[sample(1:nrow(titanic), 500), c(1,2,3,6,7,9)]
new <- titanic_small[1,]

model_titanic_glm <- glm(survived == "yes" ~ gender + age + fare + class + sibsp,
                         data = titanic_small, family = "binomial")
explain_titanic_glm <- explain(model_titanic_glm,
                               data = titanic_small[,-6],
                               label = "glm")

break_down(explain_titanic_glm, new)
```
>contribution
glm: intercept            0.298
glm: age = 50            -0.118
glm: gender = male       -0.086
glm: class = 2nd         -0.046
glm: sibsp = 0            0.005
glm: fare = 13            0.000
glm: prediction           0.053

```
break_down(model_titanic_glm,
           data =  titanic_small[,-6],
           label = "glm",
           new_observation = new)
```
> Error

```
local_attributions(model_titanic_glm,
                   data =  titanic_small[,-6],
                   label = "glm",
                   new_observation = new)
```
>contribution
glm: intercept           -1.139
glm: age = 50            -1.043
glm: gender = male       -0.613
glm: class = 2nd         -0.284
glm: sibsp = 0            0.195
glm: fare = 13            0.010
glm: prediction          -2.873

Shouldn't output be the same? Should the warning be added?

**Comment by pbiecek** [comment]
The difference between
```
break_down(explain_titanic_glm, new)
```

**Comment by pbiecek** [comment]
https://stackoverflow.com/questions/4357101/promise-already-under-evaluation-recursive-default-argument-reference-or-earlie

**Comment by hbaniecki** [comment]
There could be a warning when not using `DALEX::explain()`. 
Fixed the error with #44

**Comment by pbiecek** [comment]
Right, but then all `*.default` functions from auditor, iBreakDown, ingredients need to warn user it the default `predict()` is used. Let's discuss this f2f

**Comment by pbiecek** [comment]
And thanks for fix #44

**Comment by hbaniecki** [comment]
This was resolved with new DALEX::explain()
